lib/dispatcher.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require_relative 'commands/base'
+
+require_relative 'commands/start'
+require_relative 'commands/stop'
+require_relative 'commands/schedule'
+require_relative 'commands/speakers'
+
+class Dispatcher
+  COMMANDS = {
+    '/start' => Commands::Start,
+    '/stop' => Commands::Stop,
+    '/schedule' => Commands::Schedule,
+    '/speakers' => Commands::Speakers
+  }.freeze
+
+  def initialize(bot)
+    @bot = bot
+  end
+
+  def call(message)
+    case message
+    when Telegram::Bot::Types::Message
+      dispatch_message(message)
+    when Telegram::Bot::Types::CallbackQuery
+      dispatch_callback(message)
+    end
```

Comment:
You don't have default behavior for unknown message.

```ruby
def call(message)
  case message
  when Telegram::Bot::Types::Message
    dispatch_message(message)
  when Telegram::Bot::Types::CallbackQuery
    dispatch_callback(message)
  else 
    unknown_command(message)
  end
end

def unknown_command(message)
  bot.api.send_message(chat_id: message.chat.id, text: Locale.t('errors.unknown_command'))
end
````

Reply:
This logic is placed in `#dispatch_message`